backend/core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java

```diff
@@ -230,57 +252,74 @@ public void onConnectionCredentialsChanged(ConnectionCredentialsChangedEvent eve
-      var scopesPerProjectKey = scopesToSync.stream().collect(groupingBy(BoundScope::getSonarProjectKey, mapping(BoundScope::getConfigScopeId, toSet())));
+      scopesToSync = scopesToSync.stream()
+        .filter(boundScope -> shouldSynchronizeBinding(new Binding(connectionId, boundScope.getSonarProjectKey()))).collect(toList());
+      var scopesPerProjectKey = scopesToSync.stream()
+        .collect(groupingBy(BoundScope::getSonarProjectKey, mapping(BoundScope::getConfigScopeId, toSet())));
       scopesPerProjectKey.forEach((projectKey, configScopeIds) -> {
+        bindingSynchronizationTimestampRepository.setLastSynchronizationTimestampToNow(new Binding(connectionId, projectKey));
         LOG.debug("Synchronizing storage of Sonar project '{}' for connection '{}'", projectKey, connectionId);
         var analyzerConfigUpdateSummary = serverConnection.sync(serverApi, projectKey, cancelMonitor);
         // XXX we might want to group those 2 events under one
         if (!analyzerConfigUpdateSummary.getUpdatedSettingsValueByKey().isEmpty()) {
           applicationEventPublisher.publishEvent(
-            new SonarServerSettingsChangedEvent(configScopeIds, analyzerConfigUpdateSummary.getUpdatedSettingsValueByKey())
-          );
+            new SonarServerSettingsChangedEvent(configScopeIds, analyzerConfigUpdateSummary.getUpdatedSettingsValueByKey()));
         }
         applicationEventPublisher.publishEvent(new AnalyzerConfigurationSynchronized(configScopeIds));
         sonarProjectBranchesSynchronizationService.sync(connectionId, projectKey, cancelMonitor);
       });
       synchronizeProjectsSync(
         Map.of(connectionId, scopesToSync.stream().map(scope -> new BoundScope(scope.getConfigScopeId(), connectionId, scope.getSonarProjectKey()))
-          .collect(groupingBy(BoundScope::getSonarProjectKey, toCollection(ArrayList::new)))), cancelMonitor);
+          .collect(groupingBy(BoundScope::getSonarProjectKey, toCollection(ArrayList::new)))),
+        cancelMonitor);
     } catch (Exception e) {
       LOG.error("Error during synchronization", e);
     } finally {
       ignoreBranchEventForScopes.removeAll(scopesToSync.stream().map(BoundScope::getConfigScopeId).collect(toSet()));
     }
   }
 
+  private boolean shouldSynchronizeBinding(Binding binding) {
+    var syncPeriod = Long.parseLong(System.getProperty("sonarlint.internal.synchronization.scope.period", "300"));
+    boolean result = bindingSynchronizationTimestampRepository.getLastSynchronizationDate(binding)
+      .map(lastSync -> lastSync.isBefore(Instant.now().minus(syncPeriod, ChronoUnit.SECONDS)))
+      .orElse(true);
+    if (!result) {
+      LOG.debug("Skipping synchronization of binding '{}' because it was synchronized recently", binding);
+    }
+    return result;
+  }
+
   private boolean shouldSynchronizeScope(BoundScope configScope) {
     var syncPeriod = Long.parseLong(System.getProperty("sonarlint.internal.synchronization.scope.period", "300"));
```

Comment:
`syncPeriod` can be defined as a constant. Actually, all System properties can be defined as constants or fields to be set in post construct method.

Reply:
The properties are set from tests to influence the period, so I think they have to be read each time. Maybe we could come up with a better approach, but I will keep it for later as it's not directly related to this PR